editor.planx.uk/src/@planx/components/MultipleFileUpload/schema.test.ts

```diff
@@ -149,3 +156,97 @@ describe("multipleFileUploadSchema", () => {
     expect(result).toBe(true);
   });
 });
+
+describe("slotSchema", () => {
+  it("rejects slots with a failed upload", async () => {
+    const mockSlots = [
+      { status: "error" },
+      { status: "success" },
+    ] as FileUploadSlot[];
+
+    const result = await slotsSchema.isValid(mockSlots);
+
+    expect(result).toBe(false);
+  });
+
+  it("rejects slots which are still uploading", async () => {
+    const mockSlots = [
+      { status: "uploading" },
+      { status: "success" },
+    ] as FileUploadSlot[];
+
+    const result = await slotsSchema.isValid(mockSlots);
+
+    expect(result).toBe(false);
+  });
+
+  it("allows slots with all files uploaded", async () => {
+    const mockSlots = [
+      { status: "success" },
+      { status: "success" },
+    ] as FileUploadSlot[];
+
+    const result = await slotsSchema.isValid(mockSlots);
+
+    expect(result).toBe(true);
+  });
+});
+
+describe("fileListSchema", () => {
+  it("rejects if the proper context is not provided for validation", async () => {
+    const mockFileList = {
+      required: [],
+      recommended: [],
+      optional: [],
+    };
+    await expect(() => fileListSchema.validate(mockFileList)).rejects.toThrow(
+      /Missing context for fileListSchema/
+    );
+  });
+
+  it("rejects if any 'required' fileTypes do not have slots set", async () => {
+    const mockSlots = [{ id: "123" }, { id: "456" }] as FileUploadSlot[];
+
+    const mockFileList = {
+      required: [{ ...mockFileTypes.AlwaysRequired, slots: undefined }],
+      recommended: [],
+      optional: [],
+    };
+
+    await expect(() =>
+      fileListSchema.validate(mockFileList, { context: { slots: mockSlots } })
+    ).rejects.toThrow(/Please tag all files/);
+  });
+
+  it("rejects if any slots are untagged", async () => {
```

Comment:
Requiring _all_ uploaded files to have tags, rather than just satisfying the "required" tags is interesting / not totally intuitive to me! 

We currently allow `FileUpload` component to be created without setting a passport variable - and I'd be curious if there's any legitimate/regular uses for that in "real" content that might counteract this rule? But happy to go into testing with this assumption and see if it comes up :+1:

Reply:
This is a fair question! Here's my reasoning - 

 - Unless a file is tagged, we don't know what it is and where to put it in the passport.
 - If you upload a file on staging atm via `MultipleFileUpload` it will not get added to the passport or breadcrumbs unless it is tagged - in that context this validation makes sense.
 
You are right though, `FileUpload` files do still pass along "anonymous" files to BOPS (untagged and missing description)/Uniform/Email if we don't have a passport variable for them. It does this by assigning them to `passport.nodeId` if there isn't a value set for the component. We could implement this in `generatePayload()` and drop the validation here?

I think it's a good question to raise - let's see what come back from testing. I think we can add it to a list of questions to flush out with Alastair/POs.